README.md

```diff
@@ -35,26 +26,34 @@ Validating a Statement against a Statement Template involves three aspects:
 
-- `:predicate` - Returns `true` for a valid Statement, `false` otherwise. Default for both functions.
+- `:id` - The Statement Template ID.
+- `:predicate-fn` - A function that takes a Statement and returns `true` if that Statement is valid against it, `false` otherwise.
+- `:validator-fn` - A function that takes a Statement and returns `nil` if that Statement is valid against it, a map of error data otherwise.
 
-- `:option` - Returns the statement if it's valid, `nil` otherwise. (This is similar to the Option type in OCaml or the Just type in Haskell.)
-
-- `:result` - Returns the validation error data if the Statement is invalid, `nil` otherwise. (This is similar to the Result type in OCaml.)
+`compile-profiles->validators` takes the following keyword args:
 
-- `:assertion` - Throws an exception if the Statement is invalid, returns `nil` otherwise.
+- `:statement-ref-fns` - A map used for Statement Ref validation; if `nil`, then Statement Ref validation is ignored. See the [Statement Ref Templates](#statement-ref-templates) section for more details.
+- `:validate-profile?` - Validates the profiles against the xAPI Profile spec and checks for Profile and Template ID clashes. Default `true`.
+- `:selected-profiles` - Which Profiles in the collection should be compiled.
+- `:selected-patterns` - Which Patterns in the Profiles should be compiled. Useful for selecting only one Pattern to match against.
 
-`validation-statement-vs-template` takes an addition option for `:fn-type`:
+The `validate-statement` function take the keyword argument `:fn-type`, which can be set to the following:
 
+- `:predicate` - Returns `true` for a valid Statement, `false` otherwise. Default for both functions.
+- `:option` - Returns the statement if it's valid, `nil` otherwise. (This is similar to the Option type in OCaml or the Just type in Haskell.)
+- `:result` - Returns the validation error data if the Statement is invalid, `nil` otherwise. (This is similar to the Result type in OCaml.)
+- `:assertion` - Throws an exception if the Statement is invalid, returns `nil` otherwise.
 - `:printer` - Prints an error message when the Statement is invalid. Always returns `nil`.
+- `:templates` - Returns a vector of the IDs of the Statement Templates the Statement is valid for.
 
-`validation-statement-vs-profile` takes an additional option for `:fn-type`:
+`validate-statement` also takes the following two keyword args:
 
-- `:templates` - Returns a vector of the IDs of the Statement Templates the Statement is valid for.
+- `:all-valid?` - If `false` (default), the Statement is considered valid if _any_ of the Statement Template is valid for it. If `true`, validity is if _all_ of the Templates are valid for it. Applicable to all function types except `:templates`.
+- `:short-circuit?` - If `false` (default), returns error data for all invalid Templates; if `true`, returns data for only the first invalid Template found. Applicable to `:result`, `:assertion`, and `:printer`.
```

Comment:
These are really useful, and thoughtful to provide. The combination of the two gives lib consumers a lot of flexibility